### PR TITLE
New Block: OpenTable Reservation Form

### PR DESCRIPTION
**Pull request by madalingorbanescu**

### CODE REVIEW

File: src/blocks/opentable/save.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	} = props;
+
+	return (
+
+		<div className="nova-opentable">
+			<form method="get" className="nova-opentable__form" action="//www.opentable.com/restaurant-search.aspx" target="_blank">
+				<div className="nova-opentable__wrapper">
+					<div className="nova-opentable__date nova-opentable__input-wrap">
+						<label>{dateLabel}</label>
+						<input id="datepicker" name="startDate" className="otb-date otb-input" type="text" value="8/30/2019" readOnly="" autoComplete="off"/>
```
Comment (razwan) [inline src/blocks/opentable/save.js]: Use the date today for the initial value of the field.

File: src/blocks/opentable/scss/style.scss [hunk 2]
```diff
@@ -0,0 +1,137 @@
+.nova-opentable__wrapper {
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/scss/style.scss]: Done.

File: src/blocks/opentable/frontend.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+(function($, window, undefined) {
+
+	$( "#datepicker" ).datepicker(
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/frontend.js]: We've removed jquery-ui and now we are using <input type="date">.

File: src/blocks/opentable/frontend.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+				"August",
+				"September",
+				"October",
+				"November",
+				"December"
+			],
+			nextText: ">",
+			prevText: "<",
+			beforeShow: function() {
+				$( "#ui-datepicker-div" ).addClass( "nova-blocks-datepicker" )
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/frontend.js]: We have removed that part.

File: src/blocks/opentable/scss/style.scss [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  border: 0;
+  padding: 0;
+}
+
+.nova-blocks-datepicker a {
+  text-decoration: none;
+}
+
+.nova-blocks-datepicker .ui-datepicker-header {
+  background-color: $nova-blue;
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/scss/style.scss]: We have removed that part.

File: src/blocks/opentable/scss/style.scss [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  color: $nova-white;
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/scss/style.scss]: We have removed that part.

File: src/blocks/opentable/scss/style.scss [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+  &:not(.ui-state-active) {
+    &:hover {
+      border-color: currentColor;
+    }
+  }
+}
+
+.nova-blocks-datepicker .ui-state-active {
+  background-color: $nova-blue;
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/scss/style.scss]: We have removed that part.

File: src/blocks/opentable/scss/style.scss [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+  position: absolute;
+  top: 50%;
+  transform: translateY(-50%);
+  width: 25px;
+  height: 25px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+  color: $nova-white;
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/scss/style.scss]: We've removed that part.

File: src/blocks/opentable/save.js (hunk 1 above)
Comment (madalingorbanescu) [reply src/blocks/opentable/save.js]: Done.

File: src/blocks/opentable/scss/style.scss [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+.nova-opentable__button-wrap {
+  display: flex;
+  flex-basis: 100%;
+  align-items: center;
+  justify-content: center;
+
+  margin-top: 1em;
+}
+
+.nova-blocks-datepicker {
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/scss/style.scss]: We've have removed datepicker.

File: src/blocks/opentable/save.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+							<option value="9:30pm">9:30 pm</option>
+							<option value="10:00pm">10:00 pm</option>
+							<option value="10:30pm">10:30 pm</option>
+							<option value="11:00pm">11:00 pm</option>
+							<option value="11:30pm">11:30 pm</option>
+						</select>
+					</div>
+					<div className="nova-opentable__size nova-opentable__input-wrap">
+						<label>{partySizeLabel}</label>
+						<select id="party-size" name="partySize" className="otb-people otb-input">
```
[the remark this answers is not in the document]
Comment (madalingorbanescu) [reply src/blocks/opentable/save.js]: Done.

File: src/blocks/opentable/preview.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+		attributes: {
+			dateLabel,
+			partySizeLabel,
+			timeLabel,
+			submitButtonText,
+		},
+		setAttributes,
+	} = props;
+
+	let currentTime = new Date(),
```
Comment (razwan) [inline src/blocks/opentable/preview.js]: create a `getCurrentDate( separator )` function in `utils.js` to reduce duplicated code

File: src/blocks/opentable/save.js [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+			dateLabel,
+			timeLabel,
+			partySizeLabel,
+			submitButtonText
+		},
+	} = props;
+
+	const partySize = ( Array.from( Array( 11 ).keys() ) ).slice( 1 );
+
+	const partySizeOptions = partySize.map( option => <option value={option}> {option} </option> );
```
Comment (razwan) [inline src/blocks/opentable/save.js]: move `partySizeOptions` and `arrayTimeOptions` in `./lib/settings.php` and use `./components/with-settings` to retrieve these setttings

Comment (madalingorbanescu) [reply src/blocks/opentable/save.js]: We are not adding an option for party size and reservation time. We will let those values that are defaults in OpenTable.

File: src/blocks/opentable/preview.js (hunk 11 above)
Comment (madalingorbanescu) [reply src/blocks/opentable/preview.js]: I have added another param that will help me to change date format, if there is the case.

**Pull request merged**